cvxpy/tests/test_KKT.py

```diff
@@ -0,0 +1,301 @@
+        return sth
+
+    def test_sdp_1max(self):
+        sth = STH.sdp_1('max')
+        sth.solve(solver='SCS')
+        sth.verify_objective(places=2)
+        sth.check_primal_feasibility(places=2)
+        sth.check_complementarity(places=2)
+        sth.check_dual_domains(places=2)
+        sth.check_stationary_lagrangian(places=2)
+        return sth
+
+    def test_sdp_2(self):
+        # places is set to 3 rather than 4, because analytic solution isn't known.
+        sth = STH.sdp_2()
+        sth.solve(solver='SCS')
+        sth.verify_objective(places=2)
+        sth.check_primal_feasibility(places=2)
+        sth.verify_primal_values(places=2)
+        sth.check_complementarity(places=2)
+        sth.check_dual_domains(places=2)
+        sth.check_stationary_lagrangian(places=2)
+        return sth
+
+
+class StandardTestPCPs(BaseTest):
+
+    def test_pcp_1(self, places: int = 2):
+        sth = STH.pcp_1()
+        sth.solve(solver='SCS')
+        sth.verify_objective(places)
+        sth.check_primal_feasibility(places)
+        sth.verify_primal_values(places)
+        sth.check_complementarity(places)
+        sth.verify_dual_values(places)
+        sth.check_stationary_lagrangian(places)
+        return sth
+
+    def test_pcp_2(self, places: int = 2):
+        sth = STH.pcp_2()
+        sth.solve(solver='SCS')
+        sth.verify_objective(places)
+        sth.check_primal_feasibility(places)
+        sth.verify_primal_values(places)
+        sth.check_complementarity(places)
+        sth.verify_dual_values(places)
+        sth.check_stationary_lagrangian(places)
+        return sth
+
+    def test_pcp_3(self, places: int = 2):
+        sth = STH.pcp_3()
+        sth.solve(solver='SCS')
+        sth.verify_objective(places)
+        sth.check_primal_feasibility(places)
+        sth.verify_primal_values(places)
+        sth.check_complementarity(places)
+        sth.verify_dual_values(places)
+        sth.check_stationary_lagrangian(places)
+        return sth
+
+
+class StandardTestFlags(BaseTest):
+    """
+    A class with tests for testing out the incorporation of
+    implicit constraints within `check_stationarity_lagrangian`
+    """
+
+    @staticmethod
+    def tf_1() -> STH.SolverTestHelper:
+        """
+        Tests NSD flag
+        """
+        X = cp.Variable(shape=(3,3), NSD=True)
+        obj = cp.Maximize(cp.lambda_min(X))
+        cons = [X == X.T, X[0][1] == 123]
+        con_pairs = [
+            (cons[0], None),
+            (cons[1], None)
+        ]
+        var_pairs = [(X, np.array([[-122.99997839,  122.99999997, 0],
+                                   [ 122.99999997, -122.99997839, 0],
+                                   [ 0, 0, -111.45713829]]))]
+        obj_pair = (obj, -245.99997835783373)
+        sth = STH.SolverTestHelper(obj_pair, var_pairs, con_pairs)
+        return sth
+
+    @staticmethod
+    def tf_2() -> STH.SolverTestHelper:
+        """
+        Tests PSD flag
+        """
```

Comment:
Please say how you got the claimed optimal values.

Reply:
With what solver? What settings?

Reply:
I'll mention the solver I used, but I didn't really pass in any special solver options (as are defined [here](https://www.cvxpy.org/tutorial/advanced/index.html#setting-solver-options)).

Reply:
Please push changes so I can see changes for this point.